src/stores/appStore.ts

```diff
@@ -9,101 +8,63 @@ import {
-    },
     actions: {
         async initialiseData() {
-            this.appConfig = await getAppConfig();
-
-            // Read all adm1 indicators from a single file
-            const allIndicators = await getGlobalIndicators(1);
-            // Load all admin1 geojson - load simplified boundaries only, from a
-            // single file,
-            const allGeojson = await getGlobalGeojsonFeatures(1);
+            const state = this as AppState;
+            state.appConfig = await getAppConfig();
+            state.countryBoundingBoxes = await getCountryBoundingBoxes();
+            state.admin1Geojson = await getGlobalGeojsonFeatures(1);
+            state.admin1Indicators = await getGlobalIndicators(1);
 
-            this.countryBoundingBoxes = await getCountryBoundingBoxes();
-            this.admin1Geojson = allGeojson;
-            this.admin1Indicators = allIndicators;
+            state.initialisationComplete = true;
         },
-        async selectCountry(countryId: string) {
-            if (countryId === this.selectedCountryId) {
-                return;
-            }
+        async updateMapSettings(newMapSettings: MapSettings) {
+            const state = this as AppState;
+            const { country, adminLevel } = newMapSettings;
 
-            if (!countryId) {
-                this.selectedCountryId = "";
-                this.admin0GeojsonFeature = null;
+            if (!country) {
+                state.mapSettings = newMapSettings;
+                state.admin0GeojsonFeature = null;
                 return;
             }
 
+            let level = adminLevel;
             // Some countries do not have admin2 regions or data - if one of these is selected, we load
             // a more detailed geojson, and re-use its admin1 indicators as "admin2"
-            const level = this.appConfig.countriesWithoutAdmin2.includes(countryId) ? 1 : 2;
+            const admin2DataMissing = state.appConfig.countriesWithoutAdmin2.includes(country);
+            if (level === 2 && admin2DataMissing) {
+                level = 1;
+                state.admin2Indicators[country] = state.admin1Indicators[country];
+            }
 
-            if (!(countryId in this.admin2Indicators)) {
-                if (level === 1) {
-                    this.admin2Indicators[countryId] = this.admin1Indicators[countryId];
-                } else {
-                    this.admin2Indicators[countryId] = await getIndicators(countryId, level);
-                }
+            if (level === 2 && !admin2DataMissing && !(country in state.admin2Indicators)) {
+                state.admin2Indicators[country] = await getIndicators(country, level);
             }
-            if (!(countryId in this.admin2Geojson)) {
-                this.admin2Geojson[countryId] = await getGeojsonFeatures(countryId, level);
+
+            if (!(country in state.admin2Geojson)) {
+                state.admin2Geojson[country] = await getGeojsonFeatures(country, level);
```

Comment:
So this is *actually* going to load the more detailed admin1 geojson for countries without admin 2 , and will store it in `admin2Geojson`? And something similar is happening for the indicators - it copies the admin1 indicators into `admin2Indicators` for that country. Bit confusing. I wonder if, in another branch, we should rename `admin1` and `admin2` indicators and geojson to something like `selectedCountry...` and ... uh.. `unSelectedCountries...`.. 😆 (but not that).

Reply:
haha maybe yh, would be nice to have these names make more sense!

Reply:
I think this might have been my fault. Mea culpa. Let me know if you think of good replacement names and we can do that in another ticket. It is basically just "features to show for the selected country" and "features to show for countries which aren't selected" (and indicators).